oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,289 @@
+            probe coverage and ligation site coverage based on user defined thresholds.
+            :param probes_info: Dataframe with probe information, filtered based on sequence properties.
+            :type probes_info: pandas.DataFrame
+            :param blast_results: DataFrame with processed blast alignment search results.
+            :type blast_results: pandas.DataFrame
+            """
+            blast_results_matches = blast_results[
+                blast_results["query_gene_id"] != blast_results["target_gene_id"]
+            ]
+            blast_results_matches = blast_results_matches.merge(
+                probes_info[["probe_id", "ligation_site"]],
+                left_on="query",
+                right_on="probe_id",
+                how="inner",
+            )
+            blast_results_matches.drop(columns=["probe_id"], inplace=True)
+            blast_results_matches_filtered = []
+
+            for probe_length in range(self.probe_length_min, self.probe_length_max + 1):
+
+                min_alignment_length = probe_length * self.coverage / 100
+                blast_results_matches_probe_length = blast_results_matches[
+                    blast_results_matches.query_length == probe_length
+                ]
+
+                if self.ligation_region == 0:
+                    blast_results_matches_probe_length = (
+                        blast_results_matches_probe_length[
+                            blast_results_matches_probe_length.alignment_length
+                            > min_alignment_length
+                        ]
+                    )
+                else:
+                    blast_results_matches_probe_length = (
+                        blast_results_matches_probe_length[
+                            (
+                                blast_results_matches_probe_length.alignment_length
+                                > min_alignment_length
+                            )
+                            | (
+                                (
+                                    blast_results_matches_probe_length.query_start
+                                    < blast_results_matches_probe_length.ligation_site
+                                    - (self.ligation_region - 1)
+                                )
+                                & (
+                                    blast_results_matches_probe_length.query_end
+                                    > (
+                                        blast_results_matches_probe_length.ligation_site
+                                        + self.ligation_region
+                                    )
+                                )
+                            )
+                        ]
+                    )
+
+                blast_results_matches_filtered.append(
+                    blast_results_matches_probe_length
+                )
+
+            blast_results_matches_filtered = pd.concat(blast_results_matches_filtered)
```

Comment:
this part should all be integrated into the probe database, i.e. every probe that is filtered out get immediately removed from the probe database